event.js

```diff
@@ -0,0 +1,155 @@
+ * Keys in the desiredOrder array appear first (in that order) and any remaining keys
+ * are appended in alphabetical order.
+ *
+ * @param {Object} obj - The object whose keys are to be ordered.
+ * @returns {Object} - A new object with keys in the desired order.
+ */
+function orderObjectKeys(obj) {
+  if (typeof obj !== 'object' || obj === null || Array.isArray(obj)) {
+    return obj;
+  }
+  // Define your desired key order
+  const desiredOrder = ["Level", "Group", "Description", "Text"];
+  const ordered = {};
+
+  // Add keys in the desired order if they exist in the object
+  for (const key of desiredOrder) {
+    if (key in obj) {
+      ordered[key] = obj[key];
+    }
+  }
+
+  // Get any keys not in the desired order and sort them alphabetically
+  const remainingKeys = Object.keys(obj)
+    .filter(key => !desiredOrder.includes(key))
+    .sort();
+
+  for (const key of remainingKeys) {
+    ordered[key] = obj[key];
+  }
+  return ordered;
+}
+
+(async function main() {
+  const client = new Client('ws://127.0.0.1:17000', false);
+  try {
+    // Wide time range and large limit
+    const query = {
+      timeRangeBegin: 0,            // start from the epoch
+      timeRangeEnd:   2147483647,     // far in the future
+      codeMask:       0xFFFFFFFF,     // all codes
+      limit:          1000,           // max number of events to retrieve
+      offset:         0,
+      flags:          1               // e.g. 'NewestFirst'
+    };
+
+    const events = await client.requestEvents(query);
+
+    // -----------------------------------------------------------------------
+    // Deduplicate events:
+    // Use only the sender and the event's data (after normalizing and ordering)
+    // to build a composite key. This ignores differences in timestamp, code,
+    // or JSON key order.
+    // -----------------------------------------------------------------------
+    const seen = new Set();
+    const uniqueEvents = [];
+
+    for (const e of events) {
+      const senderPart = (typeof e.sender === 'string') ? e.sender : '';
+
+      // Process the data: normalize, then order the keys
+      let dataPart = '';
+      if (e.data) {
+        const normalizedData = normalizeValue(e.data);
+        const orderedData = orderObjectKeys(normalizedData);
+        dataPart = canonicalize(orderedData);
+      }
+
+      // Build composite key from sender and data only
+      const compositeKey = `${senderPart}-${dataPart}`;
+
+      if (!seen.has(compositeKey)) {
+        seen.add(compositeKey);
+        uniqueEvents.push(e);
+      }
+    }
+
+    // Print the final unique events, with data keys ordered for display
+    if (uniqueEvents.length === 0) {
+      console.log("No events found (after dedup).");
+    } else {
+      console.log(`Showing ${uniqueEvents.length} unique events:\n`);
+      for (const evt of uniqueEvents) {
+        console.log(`Timestamp: ${evt.timestampSec}`);
+        console.log(`Code:      ${evt.code}`);
```

Comment:
I think for usability, the client.js should contain some helper method to convert the event code to a human-readable string. Kind of like CDP Studio Event History pane does:

![image](https://github.com/user-attachments/assets/f5551115-2780-40b6-95b7-06b94902d65a)

Common CDP event codes are listed here: https://cdpstudio.com/manual/cdp/cdplogger/eventlogreader.html#cdp-event-code-flags 

But features like this can be another future pull request